logicrunner/logicrunner_test.go

```diff
@@ -1390,3 +1390,90 @@ func New(n int) (*Child, error) {
 		lr.ProcessValidationResults(context.TODO(), m)
 	}
 }
+
+func TestConstructorReturnNil(t *testing.T) {
+	if parallel {
+		t.Parallel()
+	}
+	var contractOneCode = `
+package main
+
+import (
+	"github.com/insolar/insolar/logicrunner/goplugin/foundation"
+	"github.com/insolar/insolar/application/proxy/two"
+)
+
+type One struct {
+	foundation.BaseContract
+}
+
+func (r *One) Hello() (*string, error) {
+	holder := two.New()
+	_, err := holder.AsChild(r.GetReference())
+	if err != nil {
+		return nil, err
+	}
+	ok := "all was well"
+	return &ok, nil
+}
+`
+
+	var contractTwoCode = `
+package main
+
+import (
+	"github.com/insolar/insolar/logicrunner/goplugin/foundation"
+)
+
+type Two struct {
+	foundation.BaseContract
+}
+func New() (*Two, error) {
+	return nil, nil
+}
+func (r *Two) Hello() (*string, error) {
```

Comment:
Do we actually use this method ?

Reply:
Otherwise we can't build contract. Add comment about it.

Reply:
please crete an issue about it